spec/factories/leagues.rb

```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :league, parent: :event, class: :league do
-    game_mode League.game_modes[League.game_modes.keys.sample]
+    game_mode League.game_modes[:round_robin]
```

Comment:
I do not really understand what's happening here: you set the `game_mode` of FactoryBot-created leagues to `round_robin`, but you also set it manually to the same value in `league_spec > let(:league) `

Reply:
The line in `league_spec` was from an earlier version. Should be fixed now.

Reply:
Imo there is no need to remove the random selection of a gamemode in the factory. We just have to specify that we want a specific game mode if we test for a specific gamemode.